segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/TruncateOperation.java

```diff
@@ -187,12 +188,20 @@ private CompletableFuture<Void> relocateFirstChunkIfRequired(MetadataTransaction
                         isFirstChunkRelocated = true;
                         currentMetadata = newFirstChunkMetadata;
                         currentChunkName = newChunkName;
+
+                        log.debug("{} truncate - relocated first chunk op={}, segment={}, offset={} old={} new={} relocatedBytes={} time={}.",
+                                chunkedSegmentStorage.getLogPrefix(), System.identityHashCode(this), handle.getSegmentName(),
+                                offset, oldChunkName, newChunkName, newLength, timer.getElapsedMillis());
+
                     }, chunkedSegmentStorage.getExecutor());
         }
         return CompletableFuture.completedFuture(null);
     }
 
     private CompletableFuture<Void> copyBytes(ChunkHandle writeHandle, ChunkHandle readHandle, long startOffset, long length) {
+        Preconditions.checkArgument(length <= chunkedSegmentStorage.getConfig().getMaxSizeForTruncateRelocationInbytes(),
```

Comment:
What happens if this precondition check fails? Isn't this going to relocate the chunk? what are the consequences of that?